packages/delegation-controller/src/DelegationController.test.ts

```diff
@@ -0,0 +1,634 @@
+import { SignTypedDataVersion } from '@metamask/keyring-controller';
+
+import { ROOT_AUTHORITY } from './constants';
+import {
+  DelegationController,
+  getDefaultDelegationControllerState,
+} from './DelegationController';
+import type {
+  Address,
+  Delegation,
+  DelegationControllerMessenger,
+  DelegationControllerState,
+  DelegationEntry,
+  Hex,
+} from './types';
+import { toDelegationStruct } from './utils';
+
+const FROM_MOCK = '0x2234567890123456789012345678901234567890' as Address;
+const SIGNATURE_HASH_MOCK = '0x123ABC';
+
+const CHAIN_ID_MOCK = 11155111;
+
+const VERIFYING_CONTRACT_MOCK: Address =
+  '0x0000000000000000000000000000000000000000';
+
+const DELEGATION_HASH_MOCK: Hex =
+  '0x0000000000000000000000000000000000000000000000000000000000987EDF';
+
+const DELEGATION_MOCK: Delegation = {
+  delegator: '0x1234567890123456789012345678901234567890' as Address,
+  delegate: FROM_MOCK,
+  authority: ROOT_AUTHORITY,
+  caveats: [
+    {
+      enforcer: '0x1111111111111111111111111111111111111111',
+      terms: '0x',
+      args: '0x',
+    },
+  ],
+  salt: '0x' as Hex,
+  signature: '0x',
+};
+
+const DELEGATION_ENTRY_MOCK: DelegationEntry = {
+  data: DELEGATION_MOCK,
+  chainId: 11155111, // sepolia
+  tags: [],
+};
+
+class TestDelegationController extends DelegationController {
+  public testUpdate(updater: (state: DelegationControllerState) => void) {
+    this.update(updater);
+  }
+}
+
+/**
+ * Create a mock messenger instance.
+ *
+ * @returns The mock messenger instance plus individual mock functions for each action.
+ */
+function createMessengerMock() {
+  const accountsControllerGetSelectedAccountMock = jest.fn();
+  const keyringControllerSignTypedMessageMock = jest.fn();
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const callMock = (method: string, ...args: any[]) => {
+    switch (method) {
+      case 'AccountsController:getSelectedAccount':
```

Comment:
Minor, I have been a mock obsessive in the past, but the nature of the messenger is that it's already a sort of "mock" as it's just an object that you set handlers on.

So maybe easier to maintain with a real `Messenger` instance and you `registerActionHandler` using your your `jest.fn`.

Reply:
I see, that makes sense. I've refactored that logic, so take a look in case I missed something there.